seed/seed.go

```diff
@@ -205,6 +205,18 @@ type PreseedCapable interface {
 	LoadPreseedAssertion() (*asserts.Preseed, error)
 }
 
+// Copier can be implemented by a seed that supports copying itself to a given
+// destination.
+type Copier interface {
+	Seed
+	// Copy copies the seed to the given seedDir with the label provided. If
+	// label is empty, then the label of the seed that implements Copier is
+	// used. This interface only makes sense to implement for UC20+ seeds. Copy
+	// requires you to call the methods LoadAssertions and LoadMeta first. Only
+	// the snaps for the mode that was passed to LoadMeta will be copied.
```

Comment:
I missed the last bit, that doesn't make sense, we need to copy the snaps for all the modes

Reply:
In that case, I can move the call to LoadMeta into Copy. Although, that will require the caller to call LoadMeta again, if they happened to call it previsouly with a different mode.

Reply:
I could check to see if the mode was changed from not-all to all, and then re-call LoadMeta at the end of the copy to put things back in place like they were before the method was called. I'm not sure if that is needed, though.

Reply:
That wouldn't preserve the effects of a custom `SnapHandler`, so probably not a good idea.